src/types.rs

```diff
@@ -600,6 +600,17 @@ fn rewrite_bare_fn(bare_fn: &ast::BareFnTy,
                    -> Option<String> {
     let mut result = String::with_capacity(128);
 
+    if !bare_fn.lifetimes.is_empty() {
+        result.push_str("for<");
+        // 4 = "for<".len()
+        result.push_str(&try_opt!(bare_fn.lifetimes
+                .iter()
+                .map(|l| l.rewrite(context, try_opt!(width.checked_sub(4)), offset + 4))
```

Comment:
Technically, the width should be subtracted by 5 to account for `>`. This also gives every item the same constraints, which can lead to incorrect results. I don't expect it to matter though, as this will probably do fine most of the time. Perhaps add a comment that we should use the list formatting machinery if this ever needs an upgrade?